docker/Dockerfile

```diff
@@ -3,38 +3,226 @@
+
+
+# Build OWT specific modules
+ARG OWTSERVER_REPO=https://github.com/open-webrtc-toolkit/owt-server.git
+ARG OPENH264_MAJOR=1
+ARG OPENH264_MINOR=7
+ARG OPENH264_SOVER=4
+ARG OPENH264_SOURCENAME=v${OPENH264_MAJOR}.${OPENH264_MINOR}.0.tar.gz
+ARG OPENH264_SOURCE=https://github.com/cisco/openh264/archive/v${OPENH264_MAJOR}.${OPENH264_MINOR}.0.tar.gz
+ARG OPENH264_BINARYNAME=libopenh264-${OPENH264_MAJOR}.${OPENH264_MINOR}.0-linux64.${OPENH264_SOVER}.so
+ARG OPENH264_BINARY=https://github.com/cisco/openh264/releases/download/v${OPENH264_MAJOR}.${OPENH264_MINOR}.0/${OPENH264_BINARYNAME}.bz2
+ARG LICODE_COMMIT="8b4692c88f1fc24dedad66b4f40b1f3d804b50ca"
+ARG LICODE_REPO=https://github.com/lynckia/licode.git
+ARG WEBRTC_REPO=https://github.com/open-webrtc-toolkit/owt-deps-webrtc.git
+ARG SVT_VER=v1.3.0
+ARG SVT_REPO=https://github.com/intel/SVT-HEVC.git
+ARG SERVER_PATH=/home/owt-server
+ARG OWT_SDK_REPO=https://github.com/open-webrtc-toolkit/owt-client-javascript.git
+ARG OWT_BRANCH=master
+
+RUN apt-get update && apt-get install -y -q --no-install-recommends python libglib2.0-dev rabbitmq-server mongodb libboost-thread-dev libboost-system-dev liblog4cxx-dev
+
+# 1. Clone OWT server source code 
+# 2. Clone licode source code and patch
+# 3. Clone webrtc source code and patch
+RUN git config --global user.email "you@example.com" && \
+    git config --global user.name "Your Name" && \
+    git clone --depth=1 -b ${OWT_BRANCH} ${OWTSERVER_REPO} && \
+
+    # Install node modules for owt
+    npm install -g --loglevel error node-gyp grunt-cli underscore jsdoc && \
+    cd owt-server && npm install nan && \
+
+    # Get openh264 for owt
+    cd third_party && \
+    mkdir openh264 && cd openh264 && \
+    wget ${OPENH264_SOURCE} && \
+    wget ${OPENH264_BINARY} && \
+    tar xzf ${OPENH264_SOURCENAME} openh264-${OPENH264_MAJOR}.${OPENH264_MINOR}.0/codec/api && \
+    ln -s -v openh264-${OPENH264_MAJOR}.${OPENH264_MINOR}.0/codec codec && \
+    bzip2 -d ${OPENH264_BINARYNAME}.bz2 && \
+    ln -s -v ${OPENH264_BINARYNAME} libopenh264.so.${OPENH264_SOVER} && \
+    ln -s -v libopenh264.so.${OPENH264_SOVER} libopenh264.so && \
+    echo 'const char* stub() {return "this is a stub lib";}' > pseudo-openh264.cpp && \
+    gcc pseudo-openh264.cpp -fPIC -shared -o pseudo-openh264.so 
+
+# Get licode for owt
+RUN cd ${SERVER_PATH}/third_party && git clone ${LICODE_REPO} && \
+    cd licode && \
+    git reset --hard ${LICODE_COMMIT} && \
+    git am ${SERVER_PATH}/scripts/patches/licode/*.patch
+
+# Install webrtc for owt
+RUN cd ${SERVER_PATH}/third_party && mkdir webrtc  && cd webrtc &&\
+    export GIT_SSL_NO_VERIFY=1 && \
+    git clone -b 59-server ${WEBRTC_REPO} src && \
+    ./src/tools-woogeen/install.sh && \
+    ./src/tools-woogeen/build.sh 
+
+# Install webrtc79 for owt
+RUN mkdir ${SERVER_PATH}/third_party/webrtc-m79 && \
+    cd ${SERVER_PATH}/third_party/webrtc-m79 && \
+    /bin/bash ${SERVER_PATH}/scripts/installWebrtc.sh
+
```

Comment:
It sure takes a long time to build, I guess from having two versions here. Is this so that you can choose between the two implementations? I finished building it EOD Friday but plan to look at it further today.

Reply:
We are upgrading webrtc to a new version, and now we update parts of OWT modules, so we keep both 2 versions. We will update the Dockerfile to the latest webrtc version once the update work is done.